superdesk/core/resources/service.py

```diff
@@ -237,19 +237,23 @@ async def validate_update(
             context={"use_objectid": True} if not self.config.query_objectid_as_string else {},
         )
 
-    async def create(self, docs: List[ResourceModelType]) -> List[str]:
+    async def create(self, docs: List[ResourceModelType | dict[str, Any]]) -> List[str]:
         """Creates a new resource
 
         Will automatically create the resource(s) in both Elasticsearch (if configured for this resource)
         and MongoDB.
 
-        :param docs: List of resources to create
+        :param docs: List of resources or dictionaries to create the registries
         :return: List of IDs for the created resources
+        :raises Pydantic.ValidationError: If any of the docs provided are not valid
         """
 
-        await self.on_create(docs)
         ids: List[str] = []
+
         for doc in docs:
+            if isinstance(doc, dict):
+                doc = self.get_model_instance_from_dict(doc)
+            await self.on_create([doc])
```

Comment:
With this change the `on_create` callback will only ever receive 1 item, therefor it doesn't make sense to have it's signature accept a list.

Rather than changing the signature of both `on_create` and `on_created` callbacks to receive 1 item (which would require changes in Newshub where these are used), we could instead convert the list of dicts before iterating over it.

I don't see this being much of a performance issue, as we rarely send more than 1 item to be created.

Reply:
I adjusted it. I had to do some adjustments to the types so mypy won't complain. Please take another look.